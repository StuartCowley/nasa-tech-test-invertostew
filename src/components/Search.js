import { useState } from "react";

import "../styles/Search.css";

function Search() {
  return (
    <>
      <input type="text" name="search" id="search" placeholder="Search" value="" />
    </>
  );
}

export default Search;
