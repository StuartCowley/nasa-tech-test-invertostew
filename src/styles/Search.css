.search-form {
  display: flex;
  flex-direction: row;
}

.search-form__input {
  width: 20%;
  height: 50px;
  border-radius: 11px;
  font-size: 20px;
  padding: 0 10px;
}
