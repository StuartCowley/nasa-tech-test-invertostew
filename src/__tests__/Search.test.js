import { render, screen } from "@testing-library/react";

import Search from "../components/Search";

describe("Search", () => {
  const validProps = {
    setSearchResults: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <Search setSearchResults={validProps.setSearchResults} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the input element", () => {
    render(<Search setSearchResults={validProps.setSearchResults} />);

    const input = screen.getByPlaceholderText(/search/i);

    expect(input).toBeInTheDocument();
  });

  it("renders the button", () => {
    render(<Search setSearchResults={validProps.setSearchResults} />);

    const button = screen.getByText(/search/i);

    expect(button).toBeInTheDocument();
  });
});
