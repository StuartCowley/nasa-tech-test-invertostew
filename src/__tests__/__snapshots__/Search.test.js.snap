// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<DocumentFragment>
  <input
    id="search"
    name="search"
    placeholder="Search"
    type="text"
    value=""
  />
</DocumentFragment>
`;
